feat(добавил кнопку войти): добавил событие нажатие кнопки войти

almaz-12 opened the pull request:
1. Создана компонента страницы “Логин”
2. Создана компонента страницы “Регистрация”
3. Создана компонента страницы “Карта”
4. Создана компонента страницы “Профиль”
5. В компоненте добавлены кнопки навигации: карта, профиль, выйти
6. Навигация сделана при помощи стейта
7. При сабмите пользователь попадает на страницу с картой

almaz-12 committed:
feat(добавил кнопку войти): добавил событие нажатие кнопки войти

Index24 commented on src/App.js:
```diff
@@ -12,19 +12,30 @@ const pages = {
-  navigateTo = (page) => {
-    this.setState({currentPage: page});
+  navigateTo = (page, isLogin) => {
+    this.setState({currentPage: page, isLogin: isLogin});
```
данная функция сразу отвечает за логику авторизации, меняя isLogin, а также за роутинг, меняя компоненты страниц. 
Рекомендую создать отдельную функцию logIn, которая перенаправляла бы пользователя при логине на страницу карты, а данная функция бы передавалась страницам. 
В компоненты страниц передавалась бы отдельная prop isLoggedIn, которая информировала бы, авторизован ли пользователь. Если нет, то мы бы переходили на страницу логина.

Index24 reviewed (commented):
Отличное начало!
Не смог найти логики и компоненты, отвечающей за регистрацию.

Пожелания, чтобы приложение не падало с ошибками (по желанию):
- обработать ввод логина (например, при пустых строках при нажатии на кнопку "To Come In" приложение падает)
- Отличное решение вынести компоненты в отдельные папки! Было бы круто css файлы называть с тем же регистром, что и файл компонента (как для App.css и App.js)
- Также, если разложили компоненты по папкам, то создав внутри каждой папки компонента файл index.js и написав следующую логику
```
import [компонент] from './[компонент].js
export default [компонент]
```
мы сможем импортировать файл в другие компоненты следующим образом
`import Logo from '../../components/logo'` 
вместо `import Logo from '../../components/logo/Logo'`
- Если успеешь, бонусом можно настроить commitizen. Он поможет грамотно создавать коммиты, которые следует разбивать на небольшие логические части. Создавая один большой коммит, в дальнейшем при расширении проекта, потребует более долгих затрат времени на его понимание, нежели если было бы несколько небольших коммитов, разделенных логическими группами

almaz-12 committed:
feat(regiter, login, app): добавил новый компонент регистрации

добавил новый компонент регистрации, добавил кнопку регистрации в layout компонента login

almaz-12 committed:
feat(logog.[cc[c): поменял регистр первой буквы css файлов

almaz-12 committed:
feat: добавил файлы index.js в папки компонентов

almaz-12 committed:
feat: добавил props в компоненты страниц

almaz-12 commented:
1. Создана компонента страницы “Регистрация”
2. Поменял регистр первой буквы css файлов
3. Добавил index.js в папки компонент
4. Добавил метод login, который сразу перенаправляет на страницу map
5. Добавил props в Компоненты страниц и добавил проверку на login, в случае не прохождения проверки, открыть страницу login

Index24 reviewed (approved):
Шикарно!

almaz-12 reviewed (commented):
ыва

almaz-12 commented:
фывфыв

almaz-12 reviewed (commented):
чсмч
